weather-project/src/components/Content/FavContent/FavCities.js

```diff
@@ -0,0 +1,50 @@
+import classes from "./FavCities.module.css";
+
+import { nameToCoord } from "../../../lib/api";
+import { dataOneCity } from "../../../lib/api";
+import { forecast5 } from "../../../lib/api";
+import { useDispatch } from "react-redux";
+import { currentActions } from "../../../store/current-slice";
+import { forecastActions } from "../../../store/forecast-slice";
+
+const FavCities = (props) => {
+  const { city } = props;
+  const dispatch = useDispatch();
+
+  async function onCityButtonHandler(city) {
+    const convertCity = await nameToCoord(city);
+    const lat = convertCity.lat;
+    const lon = convertCity.lon;
+    const fetchedCity = await dataOneCity(lat, lon);
+    const forecastData = await forecast5(lat, lon);
+
+    dispatch(
```

Comment:
dispatch wird nicht in componenten aufgerufen!! Immer auslagern, am besten in eine hook. So zum Beipiel

export const useGetTraining = (id) => {
  const dispatch = useDispatch();
  const { training } = useSelector((state) => state.toolbox.data);
  const { isTrainingLoading } = useSelector((state) => state.toolbox.loading);

  useEffect(() => {
    if (id && isEmpty(training)) dispatch(getTraining(id));
  }, []);

  const clearTraining = () => {
    dispatch(clearTrainingAction());
  };

  return {
    training, isTrainingLoading, clearTraining,
  };
};

Reply:
Bedeutet: jedesmal, wenn ich useGetTraining() aufrufe, wird, wenn eine id und training leer ist, der dispatch aufgerufen.
Zusaetzlich returne ich noch aus der hook eine Funktion clearTraining, welche ich dann in der Komponente einfach mit clearTraining() callen kann.

Reply:
const { clearTraining }  = useGetTraining(id);

clearTraining